### DISCUSSION TITLE
Data save strategy vs. Keepass2

### DISCUSSION THREAD

**Discussion by incansvl** [keepassxreboot/keepassxc#8291] [Q&A]
I've seen the issues raised about supporting Triggers (a la KP2) here. all closed as "no fix". The response from the KPXC devs seems clear, they don't want to do it, and for the main use case (sharing databases across devices via a file sharing arrangement) it's not necessary.

I'm curious. The KP2 documentation is quite clear that a 2-layer storage arrangement is necessary. To quote the trigger example description- "To ensure that KeePass changes are not lost by Dropbox, you need to use a second copy of the database that is just for the Dropbox synchronization. Then use a KeePass trigger to move changes between KeePass and Dropbox."

So, is there something fundamentally different about the way KPXC syncs updates with the database that avoids this problem, or is there something else going on?

I've been using KP2 for years, and could probably switch to using KPXC everywhere (including Linux) which would be nice, but I am unclear if KPXC could have a higher risk of database corruption if multiple machines are directly sharing the same file synced with one of the cloud file services, given the KPXC model seems to be "anyone can write at any time".

**Comment by droidmonkey** [comment]
Just save your database directly into the Dropbox folder on your computer and the dropbox app takes care of syncing the file. Nothing special about this.

**Reply by phoerious** [reply]
KeePass2's trigger architecture is really only necessary if you are working with many people at the same time on the same database. For everything else, a simple Dropbox share is more than enough and if something goes wrong, then Dropbox will save a conflict copy, which you can merge into your database if need be. Dropbox also saves a history of previous versions in case you ever want to go back to an older version. I've been using Nextcloud/Dropbox for syncing my database across multiple devices for years, no problems whatsoever.
